src/Microsoft.Health.Fhir.Api/Features/ActionResults/OperationOutcomeActionResult.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Net;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Api.Features.ActionResults
+{
+    /// <summary>
+    /// This action result is specifically used when we want to return an error
+    /// to the client with the appropriate OperationOutcome.
+    /// </summary>
+    public class OperationOutcomeActionResult : ActionResult
```

Comment:
Rename this to `OperationOutcomeResult` to be in-line with naming convention for other ActionResults. Or would it be better to rename `FhirResult` and `OperationResult` to `FhirActionResult` and `OperationActionResult`?